models/tridimensional/data_kleinstiver/cas9_mutants_stats.py

```diff
@@ -137,3 +137,106 @@ def main():
 
 if __name__ == '__main__':
     main()
+
+    
+def idx_freq(pam):
+    '''
+    Given a pam site returns the indices of mutation and frequency of mutation
+    occurring at said indices
+    '''
+    
+    mut_lst = []
+    for mutant in mutants_kleinstiver:
+        if mutant['pam'] == pam:
+            mut_lst.append(mutant)
+    id_dict = {}
+    for mutant in mut_lst:
+        for mutation in mutant['mutations']:
+            idx = mutation['aa_idx']
+            if idx in id_dict.keys():
+                id_dict[idx] = id_dict[idx] + 1
+            if not idx in id_dict.keys():
+                id_dict[idx] = 1
+    return id_dict
+
+def ss_freq(pam):
+    '''
+    Given a pam site returns the secondary structures present in the mutations
+    '''
+    
+    mut_lst = []
+    for mutant in mutants_kleinstiver:
+        if mutant['pam'] == pam:
+            mut_lst.append(mutant)
+    loc_dict = {}
+    for mutant in mut_lst:
+        for mutation in mutant['mutations']:
+            loc = mutation['sec_structure']
+            if loc in loc_dict.keys():
+                loc_dict[loc] = loc_dict[loc] + 1
+            if not loc in loc_dict.keys():
+                loc_dict[loc] = 1
+    return loc_dict
+
+
+def aafin_freq(pam):
+    '''
+    Given a pam site returns the final amino acid of mutation
+    '''
+    
+    mut_lst = []
+    for mutant in mutants_kleinstiver:
+        if mutant['pam'] == pam:
+            mut_lst.append(mutant)
+    aafin_dict = {}
+    for mutant in mut_lst:
+        for mutation in mutant['mutations']:
+            aafin = mutation['aa_group_mut']
+            if aafin in aafin_dict.keys():
+                aafin_dict[aafin] = aafin_dict[aafin] + 1
+            if not aafin in aafin_dict.keys():
+                aafin_dict[aafin] = 1
+    return aafin_dict
+
+
+def aachng_freq(pam):
+    '''
+    Given a pam site returns the indices and change of amino acids in each mutation in the form
+    aa_init to aa_fin at aa_idx
+    '''
+    
+    mut_lst = []
+    for mutant in mutants_kleinstiver:
+        if mutant['pam'] == pam:
+            mut_lst.append(mutant)
```

Comment:
`mut_lst = [mutant for mutant in mutants_kleinstiver if (mutant['pam'] == pam)]`

this abbreviates these 4 lines and is faster